### DISCUSSION TITLE
EmguCV Capture doesn't throw exception or error when the source is unavailable.

### DISCUSSION THREAD

**Discussion by 4sys-dev** [emgucv/emgucv#438] [Q&A]
We developed a multithread application, that used emgucv. 
The following codes can't throw exception or error when the camera is unavailable. At this time the thread died without any information :(

Capture capture = new Capture(Properties.Settings.Default.CameraAddress);

How can we get any error or exception from the Capture class?

Thanks in advance!

**Comment by emgucv** [comment]
You can check `capture.IsOpened` to make sure the camera has been correctly opened:
https://www.emgu.com/wiki/files/4.5.1/document/html/db1fad8e-1065-5269-f1d1-aaf2696cb89f.htm

**Reply by 4sys-dev** [reply]
> 
> 
> You can check `capture.IsOpened` to make sure the camera has been correctly opened:
> https://www.emgu.com/wiki/files/4.5.1/document/html/db1fad8e-1065-5269-f1d1-aaf2696cb89f.htm

Thank you for your reply!

This is the codesnipet: (this works perfectly, when the camera is available)

`try
{
VideoCapture capture = new VideoCapture("rtsp://192.168.1.128:554/mpeg4");

            var picture = capture.QueryFrame();

            CvInvoke.Resize(picture, picture, new Size(576, 324));

            var buffer = new VectorOfByte();
            CvInvoke.Imencode(".jpg", picture, buffer);  
            byte[] jpgBytes = buffer.ToArray();

            picture.Save("c://temp//test.jpg");


        }
        catch (Exception ex)
        {
            ex.ToString();
        }
`

When I unplug the network cable, the thread dies or goes to infinity timeout. I'd like to treat this situation.
I can't call the capture.IsOpened() because the thread die or long time to connect (without error, or exception):
VideoCapture capture = new VideoCapture("rtsp://192.168.1.128:554/mpeg4");

Thanks in advance

**Reply by emgucv** [reply]
What video capture backend are you using? 

If you unplug the network cable, does "capture.Grab()" still return true?

**Reply by 4sys-dev** [reply]
We use the EmguCV 4.5.1, and the previous code snipet there is in a simple c# .netFramework 4.7. console application. (poc project for testing the issue)

I can't call the capture.Grab() because the constructor isn't response, and the thread is waiting endless time...

In my opinion, this is same issue: https://github.com/emgucv/emgucv/issues/378

**Reply by emgucv** [reply]
You can also create an async task that can timeout, using instructions below:
https://devblogs.microsoft.com/pfxteam/crafting-a-task-timeoutafter-method/
The task will throw exception after certain milliseconds (a parameter) if the initialization / capture has not completed.

**Reply by abzgit** [reply]
Is there a relatively stable and short timeout to determine whether the statement “Emgu.cv.Capture = new Capture(rtspurl)” has completed execution? On my Windows 10 machine with 8GB of RAM, setting a 0.5-second delay results in unstable judgment, while a 1-second delay works reliably. But how can I ensure that this statement can be accurately and stably judged at the same speed on other machines?
